### PR TITLE
GitHub Issue NOAA-EMC/GSI#26.  Pushing back the recent developments fixes with FV3SAR GSI interface in the branch gsi_fv3reg4coldstart.

### PR DESCRIPTION
**Pull request by daprediction**
This is to push back recent fixes/developments for GSI interface for FV3SAR. Thanks to collaborators Ming Hu , Hongli Wang at GSL for pointing out the problem and reporting suspicious behavior of the codes. 
For more details see the ticket : pushing back the recent developments fixes with FV3SAR GSI interface in the branch gsi_fv3reg4coldstart #26

### CODE REVIEW

File: src/gsi/gsi_rfv3io_mod.f90 [hunk 1]
```diff
@@ -317,6 +320,8 @@ subroutine gsi_rfv3io_get_grid_specs(fv3filenamegin,ierr)
 !!!!!!! setup A grid and interpolation/rotation coeff.
     call generate_anl_grid(nx,ny,grid_lon,grid_lont,grid_lat,grid_latt)
 
+
```
Comment (WanshuWu-NOAA) [inline src/gsi/gsi_rfv3io_mod.f90:323]: remove these 2 empty lines

File: src/gsi/gsi_rfv3io_mod.f90 [hunk 2]
```diff
@@ -1931,16 +1942,16 @@ subroutine gsi_fv3ncdf_writeuv_v1(dynvars,varu,varv,mype_io,add_saved)
        deallocate(work,work_sub)
 !clt u and v would contain winds at either D-grid or A-grid
 !clt do not diretly use them in between fv3uv2eath and fv3_h_to_ll unless paying
-!attention to the actual storage layout
+!attent to the actual storage layout
```
Comment (WanshuWu-NOAA) [inline src/gsi/gsi_rfv3io_mod.f90:1945]: change by mistake?

File: src/gsi/gsi_rfv3io_mod.f90 [hunk 3]
```diff
@@ -2132,9 +2144,11 @@ subroutine gsi_fv3ncdf_writeps_v1(filename,varname,var,mype_io,add_saved)
 
        call check( nf90_put_var(gfile_loc,VarId,work_b) )
        call check( nf90_close(gfile_loc) )
-       deallocate(worka2,workb2)
+       if (allocated(worka2)) deallocate(worka2)
+       if ( allocated(workb2)) deallocate(workb2)
        deallocate(work_b,work_a,work_bi)
 
+
```
Comment (WanshuWu-NOAA) [inline src/gsi/gsi_rfv3io_mod.f90:2151]: remove

**Comment by WanshuWu-NOAA** [comment]
Mike,
Is it okay to have this line at the beginning of the "Files changed"
This commit does not belong to any branch on this repository.

**Comment by MichaelLueken-NOAA** [comment]
Hi Wan-Shu,

Where are you seeing this message?  I don't see this when I look under the Files Changed section.  Ultimately, it shouldn't be an issue, since this commit is only found in Ting's fork, not a commit in the official repository.

Mike

**Comment by WanshuWu-NOAA** [comment]
I cannot find that message now. Thanks for the answer.

**Review by MichaelLueken-NOAA — COMMENTED** [review]
Hi Ting,

I have completed my review on your changes.  The regression tests were run and the netcdf_fv3_regional test failed to reproduce.  Should all tests reproduce or is this failure expected?  The code was compiled in debug mode and three unused variables were flagged.  These variables have been noted in my review.  Please remove these unused variables.  The debug tests passed without issue.  The visual inspection of the changes was completed.  Please remove your initials from the noted comments in gsi_rfv3io_mod.f90.

Finally, I'll be unable to merge your changes into the master as the pull branch is now.  Please follow these steps to merge the current master into your branch so that I can complete your pull request:

1) git clone --recursive git@github.com:TingLei-NOAA/GSI.git update
2) cd update
3) git remote add upstream git@github.com:NOAA-EMC/GSI.git
4) git remote update
5) git rebase upstream/master
6) Correct conflicts in src/gsi/get_gefs_for_regional.f90
7) git add src/gsi/get_gefs_for_regional.f90
8) cd fix
9) git checkout rev2
10) cd ..
11) git add fix
12) git rebase --continue
13) git commit --amend
14) git push origin master --force

If you have any questions or comments, please let me know.

Mike

**Comment by daprediction** [comment]
Mike,
 Yes netcdf_fv3_regional test should fail because the fix on the treatment
of the roational wind .
Let me know if I can help to set the new regression test.
hanks a lot for your detailed instructions on what I need to do for the PR,
I will do it as soon as possible.
Regards,
Ting

**Comment by daprediction** [comment]
Hi, Mike,
it is done following your previous instructions.
Thanks.
Ting

**Comment by MichaelLueken-NOAA** [comment]
Hi Ting,

There are still three unused variables and two comments with initials that need to be addressed before I can move forward with your changes.  Please remove the variables and initials in gsi_rfv3io_mod.f90 as soon as possible.

Thanks,

Mike

**Comment by daprediction** [comment]
Hi, Mike,
My apology for missing removing the unused variables and the initials you
mentioned.
Now they are done.
Please let me  know if any problems.
Thanks
Ting

**Comment by MichaelLueken-NOAA** [comment]
The due date for the review committee has past with no feedback, so final approval was granted and the changes were merged to the master.

**Comment by daprediction** [comment]
Mike,
Thank you so much!
Really appreciate your continual help we can always depend on!
Ting

**Pull request merged**